src/pages/Models/Models.test.js

```diff
@@ -36,4 +37,28 @@ describe("Models page", () => {
       "17 models: 2 blocked, 4 alerts, 11 running"
     );
   });
+
+  it("displays correct grouping view", () => {
+    global.window = { location: { pathname: "/models?groupedby=owner" } };
+    const store = mockStore(dataDump);
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router>
+          <Models />
+        </Router>
+      </Provider>
+    );
+    expect(
+      wrapper.find(".p-model-group-toggle__button.is-selected").text()
+    ).toBe("status");
+    wrapper.find("button[value='owner']").simulate("click", {
+      target: { value: "owner" }
+    });
+    wrapper.update();
+    expect(
+      wrapper.find(".p-model-group-toggle__button.is-selected").text()
+    ).toBe("owner");
+    expect(global.window.location.search).toEqual("?groupedby=owner");
```

Comment:
Using the `MemoryRouter` above I'm not sure how you would inspect that this value gets updated... maybe you can inspect `MemoryRouter` and it'll have a `location` object?